------------------------------------------------------------
revno: 13228
revision-id: squid3@treenet.co.nz-20150828132334-3r7v4kh1lrqprrqg
parent: squid3@treenet.co.nz-20150801072237-5tzh2dczozz1e8sq
fixes bug: http://bugs.squid-cache.org/show_bug.cgi?id=3696
committer: Amos Jeffries <squid3@treenet.co.nz>
branch nick: 3.4
timestamp: Fri 2015-08-28 06:23:34 -0700
message:
  Bug 3696: crash when client delay pools are activated
------------------------------------------------------------
# Bazaar merge directive format 2 (Bazaar 0.90)
# revision_id: squid3@treenet.co.nz-20150828132334-3r7v4kh1lrqprrqg
# target_branch: http://bzr.squid-cache.org/bzr/squid3/3.4
# testament_sha1: d193928c4656282187f2a31bf66d43fd4c247177
# timestamp: 2015-08-28 13:33:44 +0000
# source_branch: http://bzr.squid-cache.org/bzr/squid3/3.4
# base_revision_id: squid3@treenet.co.nz-20150801072237-\
#   5tzh2dczozz1e8sq
# 
# Begin patch
=== modified file 'src/client_side.cc'
--- src/client_side.cc	2014-06-21 04:24:37 +0000
+++ src/client_side.cc	2015-08-28 13:23:34 +0000
@@ -3447,7 +3447,8 @@
 
             /* pools require explicit 'allow' to assign a client into them */
             if (pools[pool].access) {
-                ch.accessList = pools[pool].access;
+                cbdataReferenceDone(ch.accessList);
+                ch.accessList = cbdataReference(pools[pool].access);
                 allow_t answer = ch.fastCheck();
                 if (answer == ACCESS_ALLOWED) {
 

